packages/tenant-process/src/services/tenantService.ts

```diff
@@ -1570,6 +1577,53 @@ export function tenantServiceBuilder(
         await repository.createEvent(attributeAssignmentEvent);
       }
     },
+    async assignTenantDelegatedProducerFeature({
+      organizationId,
+      correlationId,
+      authData,
+      logger,
+    }: {
+      organizationId: TenantId;
+      correlationId: string;
+      authData: AuthData;
+      logger: Logger;
+    }): Promise<void> {
+      logger.info(
+        `Assigning delegated producer feature to tenant ${organizationId}`
+      );
+
+      assertRequesterIPAOrigin(authData);
+
+      const requesterTenant = await retrieveTenant(
+        organizationId,
+        readModelService
+      );
+
+      if (
+        requesterTenant.data.features.some(
+          (f) => f.type === "DelegatedProducer"
+        )
+      ) {
+        throw tenantAlreadyHasDelegatedProducerFeature(organizationId);
+      }
```

Comment:
If it throws exception, it could considerer a  another validator function?
```suggestion
    assertDelegationNotExists(requesterTenant.data)
```